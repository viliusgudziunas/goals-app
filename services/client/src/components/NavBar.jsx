import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const NavBar = ({ title }) => {
  const [collapsed, setCollapsed] = useState(true);

  return (
    <nav className='navbar navbar-expand-lg navbar-dark bg-dark'>
      <div className='container'>
        <span className='navbar-brand'>{title}</span>
        <button
          aria-controls='responsive-navbar-nav'
          type='button'
          aria-label='Toggle navigation'
          className={collapsed ? 'navbar-toggler collapsed' : 'navbar-toggler'}
          onClick={() => setCollapsed(!collapsed)}
        >
          <span className='navbar-toggler-icon' />
        </button>
        <div
          className={
            collapsed
              ? 'navbar-collapse collapse'
              : 'navbar-collapse collapse show'
          }
        >
          <div className='mr-auto navbar-nav'>
            <Link to='/' data-rb-event-key='/' className='nav-link'>
              Home
            </Link>
            <Link to='/about' data-rb-event-key='/about' className='nav-link'>
              About
            </Link>
            <Link to='/status' data-rb-event-key='/status' className='nav-link'>
              User Status
            </Link>
          </div>
          <div className='navbar-nav'>
            <Link
              className='nav-link'
              to='/register'
              data-rb-event-key='/register'
            >
              Register
            </Link>
            <Link className='nav-link' to='/login' data-rb-event-key='/login'>
              Log In
            </Link>
            <Link className='nav-link' to='/logout' data-rb-event-key='/logout'>
              Log Out
            </Link>
          </div>
        </div>
      </div>
    </nav>
  );
};

NavBar.propTypes = {
  title: PropTypes.string.isRequired
};

export default NavBar;
