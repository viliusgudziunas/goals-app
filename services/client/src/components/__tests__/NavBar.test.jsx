import React from 'react';
import { shallow } from 'enzyme';
import { MemoryRouter as Router } from 'react-router-dom';

import NavBar from '../NavBar';

const title = 'Test';
const theLinks = [
  { index: 0, endpoint: '/', link: 'Home' },
  { index: 1, endpoint: '/about', link: 'About' },
  { index: 2, endpoint: '/register', link: 'Register' },
  { index: 3, endpoint: '/login', link: 'Log In' }
];

describe('<NavBar />', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(
      <Router location='/'>
        <NavBar title={title} />
      </Router>
    );
    expect(wrapper.html()).toMatchSnapshot();
  });

  const wrapper = shallow(<NavBar title={title} />);
  it('should have a title', () => {
    expect(wrapper.find('span').get(0).props.children).toBe(title);
  });

  it('should have two navs', () => {
    expect(wrapper.find('.navbar-nav').length).toBe(2);
  });

  const links = wrapper.find('Link');
  it('should have four links', () => {
    expect(links.length).toBe(4);
  });

  theLinks.forEach(linkInfo => {
    it(`should have a link to ${linkInfo.link}`, () => {
      expect(links.get(linkInfo.index).props.to).toBe(linkInfo.endpoint);
      expect(links.get(linkInfo.index).props.children).toBe(linkInfo.link);
    });
  });
});
