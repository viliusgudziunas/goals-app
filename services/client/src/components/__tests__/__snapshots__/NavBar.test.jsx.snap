// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match the snapshot 1`] = `"<nav class=\\"navbar navbar-expand-lg navbar-dark bg-dark\\"><div class=\\"container\\"><span class=\\"navbar-brand\\">Test</span><button aria-controls=\\"responsive-navbar-nav\\" type=\\"button\\" aria-label=\\"Toggle navigation\\" class=\\"navbar-toggler collapsed\\"><span class=\\"navbar-toggler-icon\\"></span></button><div class=\\"navbar-collapse collapse\\"><div class=\\"mr-auto navbar-nav\\"><a data-rb-event-key=\\"/\\" class=\\"nav-link\\" href=\\"/\\">Home</a><a data-rb-event-key=\\"/about\\" class=\\"nav-link\\" href=\\"/about\\">About</a></div><div class=\\"navbar-nav\\"><a class=\\"nav-link\\" data-rb-event-key=\\"/register\\" href=\\"/register\\">Register</a><a class=\\"nav-link\\" data-rb-event-key=\\"/login\\" href=\\"/login\\">Log In</a></div></div></div></nav>"`;
